File: webapp/src/models/Module.ts
export class Module {
    baseName: string;
    typeName: string;
    author: string;
    version: string;
    isEnabled: boolean;

    constructor(baseName: string, typeName: string, author: string, version: string, isEnabled: boolean) {
        this.baseName = baseName;
        this.typeName = typeName;
        this.author = author;
        this.version = version;
        this.isEnabled = isEnabled;
    }

    public static fromJson(json: any): Module {
        return new Module(json.baseName, json.typeName, json.author, json.version, json.isEnabled);
    }
}

File: src/main/java/com/cimeyclust/ncpwebserver/models/Module.java
package com.cimeyclust.ncpwebserver.models;

public class Module {
    public String typeName;
    public String baseName;
    public String version;
    public String author;
    public boolean isEnabled;

    public Module(String typeName, String baseName, String version, String author, boolean isEnabled) {
        this.typeName = typeName;
        this.baseName = baseName;
        this.version = version;
        this.author = author;
        this.isEnabled = isEnabled;
    }
}
